lib/metasploit/framework/password_crackers/cracker.rb

```diff
@@ -403,6 +407,35 @@ def hashcat_crack_command
             cmd << ( "--hash-type=" + jtr_format_to_hashcat_format(format) )
           end
 
+          if optimize.present?
+            # https://hashcat.net/wiki/doku.php?id=frequently_asked_questions#what_is_the_maximum_supported_password_length_for_optimized_kernels
+            # Optimized Kernels has a large impact on speed.  Here are some stats from Hashcat 5.1.0:
+
+            # Kali Linux on Dell Precision M3800            
+            ## hashcat -b -w 2 -m 0
+            # * Device #1: Quadro K1100M, 500/2002 MB allocatable, 2MCU
+            # Speed.#1.........:   185.9 MH/s (11.15ms) @ Accel:64 Loops:16 Thr:1024 Vec:1
+
+            ## hashcat -b -w 2 -O -m 0
+            # * Device #1: Quadro K1100M, 500/2002 MB allocatable, 2MCU
+            # Speed.#1.........:   463.6 MH/s (8.92ms) @ Accel:64 Loops:32 Thr:1024 Vec:1
+
+            # Windows 10
+            # PS C:\hashcat-5.1.0> .\hashcat64.exe -b -O -w 2 -m 0
+            # * Device #1: GeForce RTX 2070 SUPER, 2048/8192 MB allocatable, 40MCU
+            # Speed.#1.........: 13914.0 MH/s (5.77ms) @ Accel:128 Loops:64 Thr:256 Vec:1
+
+            # PS C:\hashcat-5.1.0> .\hashcat64.exe -b -O -w 2 -m 0
+            # * Device #1: GeForce RTX 2070 SUPER, 2048/8192 MB allocatable, 40MCU
+            # Speed.#1.........: 31545.6 MH/s (10.36ms) @ Accel:256 Loops:128 Thr:256 Vec:1
+
+            # This change should result in 225%-250% speed boost at the sacrifice of some password length, which most likely
+            # wouldn't be tested inside of MSF since most users are using the MSF modules for word list and easy cracks.
+            # Anything of length where this would cut off is most likely being done independently (outside MSF)
+
+            cmd << ( "-O")
```

Comment:
There seems to be inconsistent whitespacing in this file. Did you want `("-O")` or `( "-O" )`? I see both.

Reply:
Meh.